types/testingutils/beacon_node.go

```diff
@@ -129,33 +131,33 @@ var TestingBeaconBlock = &bellatrix.BeaconBlock{
 			SyncCommitteeBits:      bitfield.NewBitvector512(),
 			SyncCommitteeSignature: phase0.BLSSignature{},
 		},
-		ExecutionPayload: &bellatrix.ExecutionPayload{
+		ExecutionPayload: &deneb.ExecutionPayload{
 			ParentHash:    phase0.Hash32{},
 			FeeRecipient:  bellatrix.ExecutionAddress{},
-			StateRoot:     phase0.Hash32{},
-			ReceiptsRoot:  phase0.Hash32{},
+			StateRoot:     phase0.Root{},
+			ReceiptsRoot:  phase0.Root{},
 			LogsBloom:     [256]byte{},
 			PrevRandao:    [32]byte{},
 			BlockNumber:   100,
 			GasLimit:      1000000,
 			GasUsed:       800000,
 			Timestamp:     123456789,
-			BaseFeePerGas: [32]byte{},
+			BaseFeePerGas: &uint256.Int{37},
 			BlockHash:     phase0.Hash32{},
 			Transactions:  Transactions,
```

Comment:
I suggest removing the `transactions` and `withdrawals` structures, along with their related components, including the custom SSZ logic and associated unnecessary tests. This simplification is feasible if we adopt the `go-eth-client` blinded block in a manner similar to how we currently use the regular block in our [`beacon_node_consts.go`](https://github.com/bloxapp/ssv-spec/blob/3d553a7de7b6ddeaf46bcc3a6d86258e20ae2730/types/testingutils/beacon_node_consts.go).

This approach is supported by our unit test which verifies that the regular block root matches the blinded block root. This can be reviewed in [`beacon_node_test.go`](https://github.com/bloxapp/ssv-spec/blob/3d553a7de7b6ddeaf46bcc3a6d86258e20ae2730/types/testingutils/beacon_node_test.go).

By adopting it, we can eliminate a significant amount of unnecessary code found in [`ssz.go`](https://github.com/bloxapp/ssv-spec/blob/3d553a7de7b6ddeaf46bcc3a6d86258e20ae2730/types/ssz.go), specifically within the `SSZTransactions` and `SSZWithdrawals` sections.
Additionally, this change allows us to remove unnecessary [SSZ tests](https://github.com/bloxapp/ssv-spec/tree/3d553a7de7b6ddeaf46bcc3a6d86258e20ae2730/types/spectest/tests/ssz).

Reply:
That will be good! Thanks!
But I think it would be better to make it in another PR. What do you think? @olegshmuelov @GalRogozinski

Reply:
> That will be good! Thanks! But I think it would be better to make it in another PR. What do you think? @olegshmuelov @GalRogozinski

Sure.
Btw the [transactions](https://github.com/bloxapp/ssv-spec/blob/49e2bdd9f013276c93355e5908506b63df79d1d9/types/testingutils/beacon_node.go#L95) func can be removed in the current PR as well

Reply:
Done